course_dashboard/answers_distribution.py

```diff
@@ -0,0 +1,178 @@
+        try:
+            answers = state["student_answers"]
+            if answers == {}: ## happen when the student submited his response but didn't answerd 
+                continue
+        except KeyError :
+            # happen when a student has loaded the problem but not yet answerd
+            continue
+
+        # get the id of a problem according to it's position in the problem_module (first problem number is 2, strange)
+        question_id = problem_module_root_id_dashed + problem_module_id + "_{}_1".format(problem_position + 1)
+        # get the response (remove 'choice_' string), only keep the response number
+        try:
+            answer = int(answers[question_id][-1:])
+        except KeyError :
+            # happen when a student has loaded the problem answer at least one question but not the one we search
+            continue
+
+        ## save the student answer in the xml
+        
+        choices = [element for element in problem.iter('choice')]
+        total_response = choices[answer].get('total_response')
+        
+        if total_response: # increment the attribute total_response
+            total_response = int(total_response) + 1 
+        else: # first student response encountered 
+            total_response = 1
+        # set the attribute 'total_response' to the correct value
+        choices[answer].set('total_response', total_response)
+
+    ## All student result have been add to total_response, now calculate percentage for each different choice
+    add_global_info_to_multiplechoice_problem(problem)
+
+def get_student_responses_to_problem_module(problem_module):
+    """ Search all problem_modules in the StudentModule model"""
+
+    course_id = problem_module.scope_ids.usage_id
+
+    problem_module_root_id_slashed = "i4x://{}/{}/problem/".format(course_id.org, course_id.course)
+    problem_module_id = problem_module.scope_ids.usage_id.name
+    problem_module_state_key = UsageKey.from_string(problem_module_root_id_slashed + problem_module_id)
+
+    # perform the request
+    problem_module_results = StudentModule.objects.filter(module_state_key=problem_module_state_key)
+    
+    return problem_module_results
+
+def add_global_info_to_multiplechoice_problem(problem):
+    total_responses_count = 0
+    
+    # count the number of responses for a problem
+    for choice in problem.iter('choice'):
+        if choice.get("total_response"):
+            total_responses_count += int(choice.get("total_response"))
+    # write total responses in multiplechoiceresponse tag
+    problem.set('total_response',total_responses_count)
+
+    # based on the total_response_count write a percentage for each choice 
+    for choice in problem.iter('choice'):
+       if choice.get("total_response"):
+           per = round(float(choice.get("total_response")) * (100.0 / total_responses_count), 2)
+           choice.set('percentage_response', per)
+       else:
+           choice.set('percentage_response', 0)
```

Comment:
for choice in problem.iter('choice'):
        total_response = int(choice.get("total_response", "0"))
        per = round(total_response * (100.0 / total_responses_count), 2)
        choice.set('percentage_response', per)